src/graph/CompactByFieldPointsNode.cpp

```diff
@@ -0,0 +1,121 @@
+// Copyright 2024 Robotec.AI
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <graph/NodesCore.hpp>
+#include <gpu/nodeKernels.hpp>
+#include <RGLFields.hpp>
+#include <repr.hpp>
+#include <graph/GraphRunCtx.hpp>
+
+void CompactByFieldPointsNode::setParameters(rgl_field_t field) { this->fieldToCompactBy = field; }
+
+void CompactByFieldPointsNode::validateImpl()
+{
+	IPointsNodeSingleInput::validateImpl();
+	// Needed to clear cache because fields in the pipeline may have changed
+	// In fact, the cache manager is no longer useful here
+	// To be kept/removed in some future refactor (when resolving comment in the `enqueueExecImpl`)
+	cacheManager.clear();
+}
+
+
+void CompactByFieldPointsNode::enqueueExecImpl()
+{
+	cacheManager.trigger();
+	inclusivePrefixSum->resize(input->getHeight() * input->getWidth(), false, false);
+	size_t pointCount = input->getWidth() * input->getHeight();
+	//TODO
+	int const* fieldToCompact = nullptr;
+	switch (fieldToCompactBy) {
```

Comment:
The template is resolved in a runtime.
Is there any other way to do this without a switch?

Reply:
You're looking for `virtual IAnyArray::ConstPtr IPointsNode::getFieldData(rgl_field_t field)`. Then you can `IAnyArray::asTyped()`.